### PR TITLE
Fixup TcpLayer deprecation macro cleanup.

### PR DESCRIPTION
**Pull request by Dimi1010**
The macro `PCPP_DEPRECATED_TCP_OPTION_TYPE` was always being undefined at the end of `TcpLayey.h` even if it was externally defined.

### CODE REVIEW

File: Packet++/header/TcpLayer.h [hunk 1]
```diff
@@ -6,6 +6,7 @@
 #include <string.h>
 
 #ifndef PCPP_DEPRECATED_TCP_OPTION_TYPE
```
Comment (tigercosmos) [inline Packet++/header/TcpLayer.h:8]: we already have `#pragma once`, so maybe we don't need `PCPP_DEPRECATED_TCP_OPTION_TYPE`?

Comment (Dimi1010) [reply Packet++/header/TcpLayer.h:8]: Wdym? They are different things?

Comment (seladb) [inline Packet++/header/TcpLayer.h:8]: Maybe we don't need `#ifndef PCPP_DEPRECATED_TCP_OPTION_TYPE`. We can assume no one will define it externally, same as we have with other macros in this project 🤔

Comment (Dimi1010) [reply Packet++/header/TcpLayer.h:8]: Well, if we are assuming that then yes, the ifndefs are superfluous.

**Review by tigercosmos — APPROVED** [review]

File: Packet++/header/TcpLayer.h
Comment (seladb) [inline Packet++/header/TcpLayer.h:1]: Maybe we should also remove `#undef PCPP_DEPRECATED_TCP_OPTION_TYPE` in the end of the file? 🤔

Comment (tigercosmos) [reply Packet++/header/TcpLayer.h:1]: only the current file uses this macro, so we shouldn't expose it.

Comment (seladb) [reply Packet++/header/TcpLayer.h:1]: yeah, I guess you're right. We can keep the `#unded`

**Review by seladb — APPROVED** [review]

**Pull request merged**